space_packet_parser/xtce/definitions.py

```diff
@@ -371,15 +371,38 @@ def _parse_parameter_set(
 
         return parameter_lookup
 
-    def parse_ccsds_packet(self,
-                           packet: packets.CCSDSPacket,
-                           *,
-                           root_container_name: Optional[str] = None) -> packets.CCSDSPacket:
+    def parse_bytes(self,
```

Comment:
We're not just parsing any bytes object though. We're parsing bytes of packet data. I'd suggest naming this `parse_packet_data`. I'm noticing that I the idiom where "packet data" is always a bytes object and the "Packet" is a dict.

Reply:
This is related to some of the other comments about inheritance, but this is just any bytes object actually! That is the point of this new function, we want to allow `bytes` as input, whatever a user wants to pass to us bytes-like.

Once we get the `bytes`, we then wrap that in our own class that has the reader methods on it. This is why I didn't do the full inheritance from `RawPacketData` elsewhere on `CCSDSPacketBytes`. I want a user to be able to make their own bytes generator, I don't want them to have to make a `RawPacketData(bytes)` generator if that makes sense. Maybe there is a better way to do this though? I don't love that we are wrapping a bytes object in another bytes-like object, but this was the quickest I came up with.

Reply:
I see how this is generic now (not CCSDS only) but why do we have the layer of abstraction of `parse_bytes` calling `parse_packet`? I don't see `parse_packet` being used anywhere else. Is there a use case for `parse_packet` that is not being called by `parse_bytes`?